dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowProcessor.java

```diff
@@ -159,6 +161,28 @@ public WorkflowProcessor(Contentlet contentlet, User firingUser) {
 		}
 	}
 
+	/**
+	 * Searches and returns a WorkflowAction using a given workflow action id, if the Processor
+	 * already have associated an action the existing action will be returned and no search will be
+	 * executed.
+	 */
+	private WorkflowAction findAction(Contentlet contentlet, User firingUser,
+			String workflowActionId)
+			throws LanguageException {
+
+		if (null == action) {
+			try {
+				action = getWorkflowAPI().findAction(workflowActionId, user);
+			} catch (Exception ex) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)